### PR TITLE
Upgrade to v2 endpoints

### PR DESCRIPTION
**Pull request by CollinEMac**
This is a pretty hefty overhaul of the python-linkedin library to make it work with LinkedIn's new V2 api. I've also removed code that Divvy doesn't use.

Also note that I didn't add any tests to this library because there wasn't any existing test suite and this code will be tested at the Divvy project level.

API docs: https://docs.microsoft.com/en-us/linkedin/

### CODE REVIEW

File: linkedin/linkedin.py [hunk 1]
```diff
@@ -170,374 +148,113 @@ def make_request(self, method, url, data=None, params=None, headers=None,
...
-        raise_for_error(response)
-        return response.json()
+    def submit_image_share(self, company_id=None, comment=None, title=None,
+                     description=None, submitted_image_url=None,
+                     visibility_code='PUBLIC'):
 
-    def submit_share(self, comment=None, title=None, description=None,
-                     submitted_url=None, submitted_image_url=None,
-                     visibility_code='anyone'):
+        if company_id is None:
```
Comment (letoss) [inline linkedin/linkedin.py]: if not company_id:

Comment (letoss) [inline linkedin/linkedin.py:199]: It's not necessary to negate the statement if you will write the _else_anyway;

```
if company_id:
    author_id = company_id
else:
    author = self.get_profile()
    author_id = author['ID']
```

**Review by letoss — APPROVED** [review]

**Comment by letoss** [comment]
We don't have the label, but consider this potato.

File: linkedin/linkedin.py [hunk 2]
```diff
@@ -170,374 +148,113 @@ def make_request(self, method, url, data=None, params=None, headers=None,
...
-        raise_for_error(response)
-        return True
-
-    def leave_group(self, group_id):
-        url = '%s/~/group-memberships/%s' % (ENDPOINTS.PEOPLE, str(group_id))
-        response = self.make_request('DELETE', url)
-        raise_for_error(response)
-        return True
+    def submit_text_share(self, company_id=None, comment=None, title=None,
+                     description=None, visibility_code='PUBLIC'):
```
Comment (ganiserb) [inline linkedin/linkedin.py]: Indentation

File: linkedin/linkedin.py [hunk 3]
```diff
@@ -170,374 +148,113 @@ def make_request(self, method, url, data=None, params=None, headers=None,
...
-        raise_for_error(response)
-        return True
-
-    def like_post(self, post_id, action):
-        url = '%s/%s/relation-to-viewer/is-liked' % (ENDPOINTS.POSTS, str(post_id))
-        try:
-            self.make_request('PUT', url, data=json.dumps(action))
-        except (requests.ConnectionError, requests.HTTPError) as error:
-            raise LinkedInError(error.message)
+        if company_id is None:
```
Comment (ganiserb) [inline linkedin/linkedin.py]:
```
author_id = company_id
if not company_id:
    author = self.get_profile()
    author_id = author['ID']
```

File: linkedin/linkedin.py (hunk 1 above)
Comment (ganiserb) [reply linkedin/linkedin.py:199]: Or this works too... Heh, interesting. We made the same suggestions.

**Review by ganiserb — APPROVED** [review]
Looks good! I wish we had tests for these new repos :man_shrugging: 

:potato: 

(To clarify, just in case. Do not add tests)

**Comment by ganiserb** [comment]
:potato:

**Comment by CollinEMac** [comment]
There's really nothing for product approval here. I'll create another PR for pointing the main Divvy project to this version of python-linkedin.

**Pull request merged**